README.md

```diff
@@ -153,46 +151,41 @@ into the element. This is the only method that must be implemented by subclasses
   the `render` implementation is a [pure function](https://en.wikipedia.org/wiki/Pure_function) using only the element's current properties
   to describe the element template. This is the only method that must be implemented by subclasses.
   Note, since `render()` is called by `update()`, setting properties does not trigger
-  `invalidate()`, allowing property values to be computed and validated.
+  an update, allowing property values to be computed and validated.
 
   * `shouldUpdate(changedProperties)` (protected): Implement to control if updating and rendering
-  should occur when property values change or `invalidate` is called. The `changedProperties`
-  argument is an object with keys for the changed properties pointing to their previous values.
+  should occur when property values change or `requestUpdate()` is called. The `changedProperties`
+  argument is a map with keys for the changed properties pointing to their previous values.
   By default, this method always returns true, but this can be customized as
   an optimization to avoid updating work when changes occur, which should not be rendered.
 
-  * `update()` (protected): This method calls `render()` and then uses `lit-html` in order to
-  render the template DOM. Implement to directly control rendered DOM.
-  Typically this is not needed as `lit-html` can be used in the `render` method
-  to set properties, attributes, and event listeners. However, it is sometimes useful
-  for calling methods on rendered elements, for example focusing an input:
-  `this.shadowRoot.querySelector('input').focus()`. The `changedProperties` argument is a Map
-  with keys for the changed properties pointing to their previous values.
-  Note, calling `super.update()` is required. Before calling `super.update()`,
-  changes made to properties do not trigger `invalidate()`, after calling `super.update()`,
-  changes do trigger `invalidate()`.
-
-  * `firstRendered()`: (protected) Called after the element's DOM has been
+  * `update(changedProperties)` (protected): This method calls `render()` and then uses `lit-html`
+  in order to render the template DOM. It also updates any reflected attributes based on
+  property values. Setting properties inside this method will *not* trigger another update..
+
+  * `firstUpdated()`: (protected) Called after the element's DOM has been
   updated the first time. This method can be useful for capturing references to rendered static
-  nodes that must be directly acted upon, for example in `update()`.
+  nodes that must be directly acted upon, for example in `updated()`. Setting properties
```

Comment:
...`updated()`, which is called immediately after?

Reply:
Done.